src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,373 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Listen,
+  Prop,
+  State,
+  h
+} from "@stencil/core";
+
+import { Component as GxComponent } from "../common/interfaces";
+
+@Component({
+  shadow: false,
+  styleUrl: "gauge.scss",
+  tag: "gx-gauge"
+})
+export class Gauge implements GxComponent {
+  @Element() element: HTMLGxGaugeElement;
+
+  /**
+   * The `gxGaugeDidLoad` event is triggered when component has been rendered completely.
+   */
+  @Event() gxGaugeDidLoad: EventEmitter;
+
+  /**
+   * Property of type Style.
+   * Define if shadow will display or not. Default is disabled.
+   */
+  @Prop() styleShadow = false;
+
+  /**
+   * This property allows selecting the gauge type. The allowed values are `circle` or `line` (defautl).
+   */
+  @Prop() type: "line" | "circle" = "line";
+
+  /**
+   *  Set `true` to display the current value. Default is `false`.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The minimum value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The current value of the gauge
+   *
+   */
+  @Prop() value: number;
+
+  /**
+   * Allows specify the width of the circumference _(When gauge is circle type)_ or the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
+
+  /**
+   * The maximum value of the gauge. _Calculated by Default_
+   */
+  @Prop() maxValue: number;
+
+  @State() rangesChildren = [];
+
+  // private totalValues = 0;
+
+  private rangesValuesAcumul = 0;
+
+  private maxValueAux = 0;
+
+  private minimumSize: number;
+
+  @Listen("gxGaugeRangeDidLoad")
+  onGaugeRangeDidLoad({ detail: childRange }) {
+    this.rangesChildren = [...this.rangesChildren, childRange];
+    this.maxValueAux += childRange.amount;
+    childRange.element.addEventListener("gxGaugeRangeDidUnload", () => {
+      this.rangesChildren = this.rangesChildren.filter(
+        elementToSave => elementToSave != childRange
+      );
+      console.log(this.rangesChildren);
+      this.maxValueAux -= childRange.amount;
+    });
+    childRange.element.addEventListener("gxGaugeRangeDidUpdate", () => {
+      const index = this.rangesChildren.findIndex(
+        elementFinding => elementFinding === childRange
+      );
+      this.rangesChildren.splice(index, 1, childRange);
+      this.maxValueAux = 0;
+      for (const childInstance of this.rangesChildren) {
+        this.maxValueAux += childInstance.amount;
+      }
+    });
+  }
```

Comment:
Please consider this alternate approach for keeping a record of the component's children: https://github.com/genexuslabs/web-controls-library/blob/51e0895e37a3086e4b63871db70a37fac2008f87/src/components/navbar/navbar.tsx#L103

Reply:
Can I add a comment in this part suggesting an upgrade with your alternate approach?
I would like to add the control to the library and that way makes it available to be used. And then make an improvement directly in this part!
Wht do you think?

Reply:
Yes, let's do that!

Reply:
Added the comment in 5db164b